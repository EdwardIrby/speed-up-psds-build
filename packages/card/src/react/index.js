import core from '@pluralsight/ps-design-system-core'
import * as glamor from 'glamor'
import glamorous from 'glamorous'
import { transparentize } from 'polished'
import React from 'react'
import Shiitake from 'shiitake'
import voidElements from 'void-elements'

const sizes = { small: 'small', medium: 'medium', large: 'large' }

const styleSize = ({ size }) =>
  ({
    [sizes.small]: {
      minWidth: 120,
      maxWidth: 240
    },
    [sizes.medium]: {
      minWidth: 240,
      maxWidth: 440
    },
    [sizes.large]: {
      minWidth: 440,
      maxWidth: 680
    }
  }[size])

const Card = glamorous.div(
  {
    width: '100%',
    textAlign: 'left'
  },
  styleSize
)

const styleOverlaysSize = ({ size }) =>
  ({
    [sizes.small]: { height: '96px' },
    [sizes.medium]: { height: '144px' },
    [sizes.large]: { height: '240px' }
  }[size])

const Overlays = glamorous.div(
  {
    position: 'relative',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    // TODO: is there a more targeted way to do this?
    ':hover div, :active div': {
      opacity: 1
    }
  },
  styleOverlaysSize
)

const ImageFrame = glamorous.div({
  position: 'relative',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  height: '100%',
  width: '100%',
  overflow: 'hidden',
  backgroundColor: core.colors.black
})

const imageStyles = glamor.css({
  position: 'relative',
  objectFit: 'cover',
  minHeight: '100%',
  minWidth: '100%'
})

const renderImage = props =>
  props.image
    ? <ImageFrame>
        {React.cloneElement(props.image, {
          ...imageStyles,
          className: props.image.props.className,
          ...(voidElements[props.image.type]
            ? {}
            : {
                children: React.Children.map(
                  props.image.props.children,
                  child =>
                    React.cloneElement(child, {
                      ...imageStyles,
                      className: child.props.className
                    })
                )
              })
        })}
      </ImageFrame>
    : null

const styleFullOverlayFullOverlayVisible = ({ fullOverlayVisible }) =>
  fullOverlayVisible ? { opacity: 1 } : {}

const FullOverlay = glamorous.div(
  {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    top: 0,
    left: 0,
    height: '100%',
    width: '100%',
    background: transparentize(0.5, core.colors.black),
    transition: `opacity ${core.motion.speedNormal}`,
    pointerEvents: 'none',
    opacity: 0
  },
  styleFullOverlayFullOverlayVisible
)

const renderFullOverlay = props =>
  props.fullOverlay
    ? <FullOverlay fullOverlayVisible={props.fullOverlayVisible}>
        {React.cloneElement(props.fullOverlay, {
          ...glamor.css({ pointerEvents: 'all' }),
          className: props.fullOverlay.props.className
        })}
      </FullOverlay>
    : null

const styleActionBarFullOverlay = ({ fullOverlay }) =>
  fullOverlay ? { background: 'none' } : {}

const styleActionBarActionBarVisible = ({ actionBarVisible }) =>
  actionBarVisible ? { opacity: 1 } : {}

const ActionBar = glamorous.div(
  {
    display: 'flex',
    justifyContent: 'flex-end',
    position: 'absolute',
    top: 0,
    right: 0,
    width: '100%',
    height: 64,
    padding: `${core.layout.spacingXSmall} ${core.layout.spacingXSmall} 0 ${core
      .layout.spacingXSmall}`,
    background: `linear-gradient(to bottom, ${transparentize(
      0.25,
      core.colors.black
    )}, transparent)`,
    transition: `opacity ${core.motion.speedNormal}`,
    pointerEvents: 'none',
    opacity: 0
  },
  styleActionBarFullOverlay,
  styleActionBarActionBarVisible
)

const ActionBarAction = glamorous.div({
  pointerEvents: 'all',
  '& + &': {
    marginLeft: core.layout.spacingTiny
  }
})

const renderActionBar = props =>
  Array.isArray(props.actionBar) && props.actionBar.length > 0
    ? <ActionBar
        actionBarVisible={props.actionBarVisible}
        fullOverlay={props.fullOverlay}
      >
        {React.Children.map(props.actionBar, (action, i) =>
          <ActionBarAction>
            {React.cloneElement(action, {
              ...glamor.css({
                ':hover, :active': {
                  background: 'none',
                  '> svg': {
                    fill: core.colors.white
                  }
                },
                '> svg': {
                  fill: transparentize(0.2, core.colors.white),
                  transition: `all ${core.motion.speedNormal}`
                }
              }),
              key: i,
              size: 'small'
            })}
          </ActionBarAction>
        )}
      </ActionBar>
    : null

const BonusBar = glamorous.div({
  position: 'absolute',
  bottom: core.layout.spacingSmall,
  left: core.layout.spacingSmall
})

const renderBonusBar = props =>
  props.bonusBar
    ? <BonusBar>
        {props.bonusBar}
      </BonusBar>
    : null

const Tag = glamorous.div({
  display: 'flex',
  justifyContent: 'flex-start',
  alignItems: 'center',
  position: 'absolute',
  top: core.layout.spacingSmall,
  left: 0,
  padding: `${core.layout.spacingTiny} ${core.layout.spacingXSmall}`,
  background: core.colors.gray04,
  borderRadius: '0 2px 2px 0'
})

const isNativeElement = el => el && typeof el.type === 'string'

const isAnchorElement = el => isNativeElement(el) && el.type === 'a'

const renderTag = props =>
  props.tag && props.size !== 'small'
    ? <Tag>
        {React.Children.map(props.tag, (part, i) =>
          React.cloneElement(part, {
            ...glamor.css({
              color: core.colors.whitek,
              textTransform: 'uppercase',
              fontSize: '10px',
              lineHeight: '16px',
              '& + &': {
                marginLeft: core.layout.spacingXSmall
              }
            }),
            className: part.props.className,
            key: i,
            size: 'tiny'
          })
        )}
      </Tag>
    : null

const Progress = glamorous.div({
  position: 'absolute',
  left: 0,
  bottom: 0,
  width: '100%',
  height: 5,
  borderTop: `1px solid ${transparentize(0.8, core.colors.black)}`,
  backgroundColor: transparentize(0.6, core.colors.gray01),
  overflow: 'hidden'
})

const percent = num => {
  try {
    return Math.min(parseFloat(num).toFixed(), 100) + '%'
  } catch (_) {
    return '0%'
  }
}

const styleProgressBarProgress = ({ progress }) => ({
  backgroundColor: percent(progress) == '100%'
    ? core.colors.green
    : core.colors.white,
  width: percent(progress)
})

const ProgressBar = glamorous.div(
  {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '0%',
    height: '5px'
  },
  styleProgressBarProgress
)

const renderProgress = props =>
  props.progress
    ? <Progress>
        <ProgressBar
          progress={props.progress}
          aria-label={`${percent(props.progress)} complete`}
        />
      </Progress>
    : null

const styleTitle = ({ size }) =>
  glamor.css(
    {
      display: 'block',
      marginTop: core.layout.spacingXSmall,
      fontWeight: core.type.fontWeightMedium,
      overflow: 'hidden',
      color: core.colors.white
    },
    {
      small: {
        fontSize: core.type.fontSizeXSmall,
        lineHeight: core.lineHeightTight,
        maxHeight: `calc(2 * ${core.type.lineHeightTight})`
      },
      medium: {
        fontSize: core.type.fontSizeSmall,
        lineHeight: core.lineHeightTight,
        maxHeight: `calc(2 * ${core.type.lineHeightTight})`
      },
      large: {
        fontSize: core.type.fontSizeMedium,
        lineHeight: core.lineHeightStandard,
        maxHeight: `calc(2 * ${core.type.lineHeightStandard})`
      }
    }[size]
  )

const linkStyles = glamor.css({
  color: 'inherit',
  cursor: 'pointer',
  textDecoration: 'none',
  ':hover, :active': {
    color: core.colors.white,
    textDecoration: 'underline',
    transition: `all ${core.motion.speedNormal}`
  }
})

const renderTitle = props =>
  isNativeElement(props.title)
    ? React.cloneElement(props.title, {
        ...styleTitle(props),
        ...linkStyles,
        className: props.title.props.className,
        children: (
          <Shiitake lines={2}>
            {props.title.props.children}
          </Shiitake>
        )
      })
    : <Shiitake lines={2} className={styleTitle(props).toString()}>
        {props.title}
      </Shiitake>

const getMetaDataLinkClassName = (props, el) =>
  classNames({
    [el.props.className]: el.props.className,
    [props.css['ps-card__metadata__datum--link']]: isAnchorElement(el)
  })

const styleMetadataSize = ({ size }) =>
  ({
    small: { fontSize: core.type.fontSizeXSmall },
    medium: { fontSize: core.type.fontSizeXSmall },
    large: { fontSize: core.type.fontSizeSmall }
  }[size])

const Metadata = glamorous.div(
  {
    display: 'flex',
    alignItems: 'center',
    fontWeight: core.type.fontWeightBook,
    lineHeight: core.type.lineHeightTight,
    color: core.colors.gray02,
    maxWidth: '100%'
  },
  styleMetadataSize
)

const MetadataDatum = glamorous.span({
  whiteSpace: 'nowrap',
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  flexShrink: 2,
  ':nth-of-type(1)': {
    flexShrink: 1
  }
})

const MetadataDot = glamorous.span({
  display: 'inline-block',
  margin: `0 ${core.layout.spacingXSmall}`
})

const renderMetaData = (props, metadata) =>
  metadata
    ? <Metadata size={props.size}>
        {metadata.map((m, i) => [
          isNativeElement(m)
            ? React.cloneElement(m, {
                ...(isAnchorElement(m) ? linkStyles : {}),
                className: m.props.className
              })
            : <MetadataDatum key={`datum${i}`}>
                {m}
              </MetadataDatum>,
          i < metadata.length - 1 &&
            <MetadataDot aria-hidden={true} key={`dot${i}`}>
              Â·
            </MetadataDot>
        ])}
      </Metadata>
    : null

const rmNonHtmlProps = props => {
  const {
    actionBar,
    actionBarVisible,
    bonusBar,
    fullOverlay,
    fullOverlayVisible,
    image,
    metadata1,
    metadata2,
    progress,
    size,
    tag,
    title,
    ...rest
  } = props
  return rest
}

const CardComponent = props =>
  <Card {...rmNonHtmlProps(props)} size={props.size}>
    <Overlays size={props.size}>
      {renderImage(props)}
      {renderFullOverlay(props)}
      {renderActionBar(props)}
      {renderBonusBar(props)}
      {renderTag(props)}
      {renderProgress(props)}
    </Overlays>
    {renderTitle(props)}
    {renderMetaData(props, props.metadata1)}
    {renderMetaData(props, props.metadata2)}
  </Card>

import PropTypes from 'prop-types'
// TODO: offer more specific guides where real constraints exist (component types)
CardComponent.propTypes = {
  actionBar: PropTypes.arrayOf(PropTypes.node),
  actionBarVisible: PropTypes.bool,
  bonusBar: PropTypes.node,
  fullOverlay: PropTypes.element,
  fullOverlayVisible: PropTypes.bool,
  image: PropTypes.element.isRequired,
  metadata1: PropTypes.arrayOf(PropTypes.node),
  metadata2: PropTypes.arrayOf(PropTypes.node),
  progress: PropTypes.number,
  tag: PropTypes.arrayOf(PropTypes.element),
  title: PropTypes.node.isRequired,
  size: PropTypes.oneOf(Object.keys(sizes))
}
CardComponent.defaultProps = {
  actionBarVisible: false,
  fullOverlayVisible: false,
  size: 'medium'
}
CardComponent.sizes = sizes

export default CardComponent
