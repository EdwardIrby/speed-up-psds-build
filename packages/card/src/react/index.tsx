import polyfillFocusWithin from 'focus-within'

import { sizes as iconSizes } from '@pluralsight/ps-design-system-icon'
import {
  names as themeNames,
  useTheme
} from '@pluralsight/ps-design-system-theme'
import {
  omit,
  RefForwardingComponent,
  ValueOf
} from '@pluralsight/ps-design-system-util'
import { compose, css, StyleAttribute } from 'glamor'
import * as React from 'react'
import Shiitake from 'shiitake'

import stylesheet from '../css'
import { toPercentageString } from '../js'
import * as vars from '../vars'

// TODO: use standard env check
if (typeof window !== 'undefined') polyfillFocusWithin(document)

type StyleFn = (
  // TODO: specify
  props?: any,
  themeName?: ValueOf<typeof themeNames>
) => StyleAttribute
const styles: { [key: string]: StyleFn } = {
  actionBar: ({ actionBarVisible: visible, fullOverlay }) => {
    const label = 'psds-card__action-bar'

    return compose(
      css(stylesheet[`.${label}`]),
      fullOverlay &&
        !visible &&
        css(stylesheet[`.${label}--fullOverlay.${label}--no-actionBarVisible`]),
      visible && css(stylesheet[`${label}--actionBarVisible`])
    )
  },
  actionButton: ({ disabled }) => {
    const label = 'psds-card__action-bar__button'

    return compose(
      css(stylesheet[`.${label}`]),
      disabled && css(stylesheet[`.${label}--disabled`])
    )
  },

  bonusBar: () => css(stylesheet['.psds-card__bonus-bar']),

  card: () => css(stylesheet['.psds-card']),

  fullOverlay: ({ fullOverlayVisible: visible }) => {
    const label = 'psds-card__full-overlay'

    return compose(
      css(stylesheet[`.${label}`]),
      visible && css(stylesheet[`.${label}--fullOverlayVisible`])
    )
  },
  fullOverlayLink: () => css(stylesheet['.psds-card__full-overlay-link']),

  image: () => css(stylesheet['.psds-card__image']),
  imageLink: () => css(stylesheet['.psds-card__image-link']),

  metadata: ({ size }, themeName) => {
    const label = 'psds-card__metadata'

    return compose(
      css(stylesheet[`.${label}`]),
      css(stylesheet[`.${label}--size-${size}`]),
      css(stylesheet[`.${label}--theme-${themeName}`])
    )
  },
  metadataDatum: () => css(stylesheet['.psds-card__metadata__datum']),
  metadataDot: () => css(stylesheet['.psds-card__metadata__dot']),

  overlays: ({ size }) => {
    const label = 'psds-card__overlays'

    return compose(
      css(stylesheet[`.${label}`]),
      css(stylesheet[`.${label}--size-${size}`])
    )
  },

  progress: () => css(stylesheet['.psds-card__progress']),
  progressBar: ({ progress }) => {
    const label = 'psds-card__progress__bar'
    const percent = toPercentageString(progress)
    const isCompleted = percent === '100%'

    return compose(
      css(stylesheet[`.${label}`]),
      isCompleted && css(stylesheet[`.${label}--complete`]),
      css({ width: percent })
    )
  },

  tag: () => css(stylesheet['.psds-card__tag']),
  tagIcon: () => css(stylesheet['.psds-card__tag__icon']),
  tagText: () => css(stylesheet['.psds-card__tag__text']),

  textLink: (_props, themeName) => {
    const label = 'psds-card__text-link'

    return compose(
      css(stylesheet[`.${label}`]),
      css(stylesheet[`.${label}--theme-${themeName}`])
    )
  },

  title: (_props, themeName) => {
    const label = 'psds-card__title'

    return compose(
      css(stylesheet[`.${label}`]),
      css(stylesheet[`.${label}--theme-${themeName}`])
    )
  },

  titleContainer: ({ size }) => {
    const label = 'psds-card__title-container'

    return compose(
      css(stylesheet[`.${label}`]),
      css(stylesheet[`.${label}--size-${size}`])
    )
  }
}

interface ActionBarActionProps extends React.HTMLAttributes<HTMLButtonElement> {
  icon?: React.ReactElement // TODO: retype as Icon when it's TS
  title: string
}
interface ActionBarActionStatics {}
interface ActionBarActionComponent
  extends RefForwardingComponent<
    ActionBarActionProps,
    HTMLButtonElement,
    ActionBarActionStatics
  > {}
const ActionBarAction = React.forwardRef((props, ref) => {
  const { title, icon, ...rest } = props
  const ariaLabel = props['aria-label'] || title
  return (
    <button
      {...styles.actionButton(props)}
      {...rest}
      aria-label={ariaLabel}
      ref={ref}
      title={title}
    >
      {icon}
    </button>
  )
}) as ActionBarActionComponent
ActionBarAction.displayName = 'Card.Action'

const FullOverlayLink: React.FC<React.HTMLAttributes<
  HTMLSpanElement
>> = props => <span {...styles.fullOverlayLink(props)} {...props} />

FullOverlayLink.displayName = 'Card.FullOverlayLink'

interface ImageProps extends React.HTMLAttributes<HTMLDivElement> {
  src: string
}
const Image: React.FC<ImageProps> = props => {
  const { src, ...rest } = props
  return (
    <div
      {...styles.image(props)}
      {...rest}
      style={{ backgroundImage: `url(${src})` }}
    />
  )
}
Image.displayName = 'Card.Image'

const ImageLink: React.FC<React.HTMLAttributes<HTMLSpanElement>> = props => (
  <span {...styles.imageLink(props)} {...props} />
)
ImageLink.displayName = 'Card.ImageLink'

interface TagProps extends React.HTMLAttributes<HTMLDivElement> {
  icon?: React.ReactElement // TODO: Icon when Icon is TS
}
const Tag: React.FC<TagProps> = ({ children, icon, ...rest }) => (
  <div {...styles.tag()} {...rest}>
    {icon && (
      <div {...styles.tagIcon()}>
        {React.cloneElement(icon, { size: iconSizes.small })}
      </div>
    )}

    <span {...styles.tagText()}>{children}</span>
  </div>
)

Tag.displayName = 'Card.Tag'

const Text: React.FC<React.HTMLAttributes<HTMLSpanElement>> = props => (
  <span {...props} />
)
Text.displayName = 'Card.Text'

const TextLink: React.FC<React.HTMLAttributes<HTMLSpanElement>> = props => {
  const themeName = useTheme()
  return <span {...styles.textLink(undefined, themeName)} {...props} />
}
TextLink.displayName = 'Card.TextLink'

const Title: React.FC<React.HTMLAttributes<HTMLDivElement>> = props => {
  const themeName = useTheme()
  const { children, ...rest } = props

  return (
    <div {...styles.title(undefined, themeName)} {...rest}>
      <Shiitake lines={2}>{children}</Shiitake>
    </div>
  )
}
Title.displayName = 'Card.Title'

interface MetaDataProps {
  metadata: (string | React.ReactElement<typeof TextLink>)[]
  size: ValueOf<typeof vars.sizes>
}
const MetaData: React.FC<MetaDataProps> = props => {
  const { metadata, size, ...rest } = props
  const themeName = useTheme()
  return (
    <div {...styles.metadata(props, themeName)} {...rest}>
      {metadata.map((m, i) => [
        <span key={`datum${i}`} {...styles.metadataDatum()}>
          {m}
        </span>,
        i < metadata.length - 1 && (
          <span aria-hidden key={`dot${i}`} {...styles.metadataDot()}>
            Â·
          </span>
        )
      ])}
    </div>
  )
}

interface CardComponentProps extends Record<string, unknown> {
  actionBar?: React.ReactElement<typeof ActionBarAction>[]
  actionBarVisible?: boolean
  bonusBar?: React.ReactNode
  fullOverlay?: React.ReactElement<typeof FullOverlayLink>
  fullOverlayVisible?: boolean
  image: React.ReactElement<typeof ImageLink>
  metadata1?: (string | React.ReactElement<typeof TextLink>)[]
  metadata2?: (string | React.ReactElement<typeof TextLink>)[]
  progress?: number
  size?: ValueOf<typeof vars.sizes>
  tag?: React.ReactElement<typeof Tag>
  title: React.ReactElement<typeof TextLink> | React.ReactElement<typeof Title>
}

interface CardComponentStatics {
  Action: typeof ActionBarAction
  FullOverlayLink: typeof FullOverlayLink
  Image: typeof Image
  ImageLink: typeof ImageLink
  sizes: typeof vars.sizes
  Tag: typeof Tag
  Text: typeof Text
  TextLink: typeof TextLink
  Title: typeof Title
}
const CardComponent: React.FC<CardComponentProps> &
  CardComponentStatics = props => {
  const htmlProps = omit<
    CardComponentProps,
    [
      'actionBar',
      'actionBarVisible',
      'bonusBar',
      'fullOverlay',
      'fullOverlayVisible',
      'image',
      'metadata1',
      'metadata2',
      'progress',
      'size',
      'tag',
      'title'
    ]
  >(props, [
    'actionBar',
    'actionBarVisible',
    'bonusBar',
    'fullOverlay',
    'fullOverlayVisible',
    'image',
    'metadata1',
    'metadata2',
    'progress',
    'size',
    'tag',
    'title'
  ])
  const size = props.size || sizes.medium
  return (
    <div {...styles.card()} {...htmlProps}>
      <div {...styles.overlays({ size })}>
        {props.image ? props.image : null}

        {props.fullOverlay ? (
          <div {...styles.fullOverlay(props)}>{props.fullOverlay}</div>
        ) : null}

        {Array.isArray(props.actionBar) && props.actionBar.length > 0 ? (
          <div {...styles.actionBar(props)}>{props.actionBar}</div>
        ) : null}

        {props.bonusBar ? (
          <div {...styles.bonusBar(props)}>{props.bonusBar}</div>
        ) : null}

        {props.tag}

        {props.progress ? (
          <div {...styles.progress(props)}>
            <div
              {...styles.progressBar(props)}
              aria-label={`${toPercentageString(props.progress)} complete`}
            />
          </div>
        ) : null}
      </div>

      <div {...styles.titleContainer({ size })}>{props.title}</div>
      {props.metadata1 && <MetaData size={size} metadata={props.metadata1} />}
      {props.metadata2 && <MetaData size={size} metadata={props.metadata2} />}
    </div>
  )
}

CardComponent.Action = ActionBarAction
CardComponent.FullOverlayLink = FullOverlayLink
CardComponent.Image = Image
CardComponent.ImageLink = ImageLink
CardComponent.Tag = Tag
CardComponent.Text = Text
CardComponent.TextLink = TextLink
CardComponent.Title = Title

CardComponent.sizes = vars.sizes

export const sizes = vars.sizes

export default CardComponent
