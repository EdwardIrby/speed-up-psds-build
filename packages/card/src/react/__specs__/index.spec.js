import Icon from '@pluralsight/ps-design-system-icon/react.js'
import React from 'react'
import { render } from '@testing-library/react'
import { shallow } from 'enzyme'

import Card from '../index.js'

test('click on Card.Title triggered once', () => {
  let callCount = 0
  const onClick = _ => {
    callCount += 1
  }
  const title = shallow(<Card.Title onClick={onClick}>Clicks once</Card.Title>)
  title.simulate('click')
  expect(callCount).toBe(1)
})

describe('#Card.Action', () => {
  it('forwards ref', () => {
    const refToForward = React.createRef()

    const { getByTestId } = render(
      <Card.Action
        data-testid="under-test"
        ref={refToForward}
        title="Some title"
        icon={<Icon id={Icon.ids.more} />}
      ></Card.Action>
    )

    const el = getByTestId('under-test')
    expect(el).toBe(refToForward.current)
  })
})
