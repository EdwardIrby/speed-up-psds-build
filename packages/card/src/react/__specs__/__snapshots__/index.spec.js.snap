// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bonusBar 1`] = `
<div
  className="css-19wk7ys"
>
  <div
    className="css-iwdsn5"
  >
    <div
      className="css-1r7bzjg"
    >
      <div
        className={undefined}
        data-css-jpg3o=""
      />
    </div>
    <div
      className="css-1nqeg2a"
    >
      <div>
        Anything
      </div>
    </div>
  </div>
  <div
    className="css-8n0rg6"
  >
    <span
      style={
        Object {
          "display": "block",
          "maxHeight": "0px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "display": "block",
            "width": "100%",
          }
        }
      >
        
        …
      </span>
      <span
        style={
          Object {
            "display": "block",
            "height": "0px",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        The card title
      </span>
      <span
        style={
          Object {
            "display": "block",
            "left": "-20000px",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
        </span>
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          
          …
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`renders custom metadata classes 1`] = `
<div
  className="css-19wk7ys"
>
  <div
    className="css-iwdsn5"
  >
    <div
      className="css-1r7bzjg"
    >
      <div
        className={undefined}
        data-css-jpg3o=""
      />
    </div>
  </div>
  <div
    className="css-8n0rg6"
  >
    <span
      style={
        Object {
          "display": "block",
          "maxHeight": "0px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "display": "block",
            "width": "100%",
          }
        }
      >
        
        …
      </span>
      <span
        style={
          Object {
            "display": "block",
            "height": "0px",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        The card title
      </span>
      <span
        style={
          Object {
            "display": "block",
            "left": "-20000px",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
        </span>
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          
          …
        </span>
      </span>
    </span>
  </div>
  <div
    className="css-bcrrqv"
  >
    <span
      className="keepme"
    >
      Custom
    </span>
  </div>
</div>
`;

exports[`renders fullOverlay 1`] = `
<div
  className="css-19wk7ys"
>
  <div
    className="css-iwdsn5"
  >
    <div
      className="css-1r7bzjg"
    >
      <div
        className={undefined}
        data-css-jpg3o=""
      />
    </div>
    <div
      className="css-1opv87a"
    >
      <div
        className={undefined}
        data-css-1fxa3fl=""
      >
        FakePlayBtn
      </div>
    </div>
  </div>
  <div
    className="css-8n0rg6"
  >
    <span
      style={
        Object {
          "display": "block",
          "maxHeight": "0px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "display": "block",
            "width": "100%",
          }
        }
      >
        
        …
      </span>
      <span
        style={
          Object {
            "display": "block",
            "height": "0px",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        The card title
      </span>
      <span
        style={
          Object {
            "display": "block",
            "left": "-20000px",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
        </span>
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          
          …
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`renders link class on meta only if anchor tag 1`] = `
<div
  className="css-19wk7ys"
>
  <div
    className="css-iwdsn5"
  >
    <div
      className="css-1r7bzjg"
    >
      <div
        className={undefined}
        data-css-jpg3o=""
      />
    </div>
  </div>
  <div
    className="css-8n0rg6"
  >
    <span
      style={
        Object {
          "display": "block",
          "maxHeight": "0px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "display": "block",
            "width": "100%",
          }
        }
      >
        
        …
      </span>
      <span
        style={
          Object {
            "display": "block",
            "height": "0px",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        The card title
      </span>
      <span
        style={
          Object {
            "display": "block",
            "left": "-20000px",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
        </span>
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          
          …
        </span>
      </span>
    </span>
  </div>
  <div
    className="css-bcrrqv"
  >
    <span
      className={undefined}
    >
      No link class
    </span>
    <span
      aria-hidden={true}
      className="css-kykfz1"
    >
      ·
    </span>
    <a
      className={undefined}
      data-css-ohpd3v=""
    >
      yes link class
    </a>
  </div>
</div>
`;

exports[`renders no tag if size is small 1`] = `
<div
  className="css-1hal8pq"
>
  <div
    className="css-b22ep1"
  >
    <div
      className="css-1r7bzjg"
    >
      <div
        className={undefined}
        data-css-jpg3o=""
      />
    </div>
  </div>
  <div
    className="css-aoggyi"
  >
    <span
      style={
        Object {
          "display": "block",
          "maxHeight": "0px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "display": "block",
            "width": "100%",
          }
        }
      >
        
        …
      </span>
      <span
        style={
          Object {
            "display": "block",
            "height": "0px",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        The card title
      </span>
      <span
        style={
          Object {
            "display": "block",
            "left": "-20000px",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
        </span>
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          
          …
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`renders required props 1`] = `
<div
  className="css-19wk7ys"
>
  <div
    className="css-iwdsn5"
  >
    <div
      className="css-1r7bzjg"
    >
      <div
        className={undefined}
        data-css-jpg3o=""
      />
    </div>
  </div>
  <div
    className="css-8n0rg6"
  >
    <span
      style={
        Object {
          "display": "block",
          "maxHeight": "0px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "display": "block",
            "width": "100%",
          }
        }
      >
        
        …
      </span>
      <span
        style={
          Object {
            "display": "block",
            "height": "0px",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        The card title
      </span>
      <span
        style={
          Object {
            "display": "block",
            "left": "-20000px",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
        </span>
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          
          …
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`renders tag 1`] = `
<div
  className="css-19wk7ys"
>
  <div
    className="css-iwdsn5"
  >
    <div
      className="css-1r7bzjg"
    >
      <div
        className={undefined}
        data-css-jpg3o=""
      />
    </div>
    <div
      className="css-1gmxx97"
    >
      <div
        className={undefined}
        data-css-1c199zc=""
        size="xSmall"
      >
        FakeSVG
      </div>
      <span
        className={undefined}
        data-css-1c199zc=""
        size="xSmall"
      >
        Course
      </span>
    </div>
  </div>
  <div
    className="css-8n0rg6"
  >
    <span
      style={
        Object {
          "display": "block",
          "maxHeight": "0px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "display": "block",
            "width": "100%",
          }
        }
      >
        
        …
      </span>
      <span
        style={
          Object {
            "display": "block",
            "height": "0px",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        The card title
      </span>
      <span
        style={
          Object {
            "display": "block",
            "left": "-20000px",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            W
          </span>
        </span>
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          
          …
        </span>
      </span>
    </span>
  </div>
</div>
`;
